apiserver/facades/agent/uniter/uniter.go

```diff
@@ -2688,7 +2703,16 @@ func (u *UniterAPI) goalStateRelations(appName, principalName string, allRelatio
 				}
 			}
 
-			result[e.Name] = relationGoalState
+			// Merge in the goal state for the current remote endpoint
+			// with any other goal state already collected for the local endpoint.
+			unitsGoalState := result[resultEndpointName]
+			if unitsGoalState == nil {
+				unitsGoalState = params.UnitsGoalState{}
+			}
+			for k, v := range relationGoalState {
+				unitsGoalState[k] = v
+			}
+			result[resultEndpointName] = unitsGoalState
```

Comment:
Since none of the tests changed other than by relation name, what impact is this supposed to have?

Reply:
in case there is more than one remote charm related to any given endpoint

Reply:
i enhanced TestGoalStatesMultipleRelations